src/datasets/user-wgn.dataset.ts

```diff
@@ -0,0 +1,176 @@
+import { Dataset, LineChartQueryOptions, LineChartMeta } from './types';
+import { xBetweenDates } from './utils';
+import { ResultRow } from '../models/data-cube/types';
+import {
+  activeUserMeasure,
+  browserCategory,
+  countryCategory,
+  eventCountMeasure,
+  revenueMeasure,
+  sourceCategory,
+} from '../models/data-cube/presets';
+import { inOneOfDateRanges } from '../models/data-cube/filters';
+import { generateCube } from 'src/models/data-cube/generation';
+import { DAY } from '../utils/timeUnits';
+import {
+  CompoundMeasure,
+  CompoundMeasureType,
+  destructureCompoundMeasure,
+  isCompoundMeasure,
+  namePeriodOverPeriodMeasure,
+  nameRollingMeasure,
+  PeriodOverPeriodMeasure,
+  RollingMeasure,
+} from '../utils/compoundMeasures';
+
+
+export interface UserWGNDatasetConfig {
+  avgHits: number;
+  hitStdDev: number;
+  avgUsers: number;
+  userStdDev: number;
+  avgSessionsPerUser: number;
+  sessionsPerUserStdDev: number;
+}
+
+export const UserWGNDatasetDefaultConfig: UserWGNDatasetConfig = {
+  avgHits: 10000,
+  hitStdDev: 100,
+  avgUsers: 100,
+  userStdDev: 1,
+  avgSessionsPerUser: 5,
+  sessionsPerUserStdDev: 3,
+};
+
+// TODO: Refactoring
+export function UserWGNDataset(config: UserWGNDatasetConfig): Dataset {

```

Comment:
What does WGN stand for?

Reply:
White Gaussian Noise. I admit that it is a bad name, and we should rename it. Any good ideas?

Reply:
`UserWhiteGaussianNoiseDataset` 😬 sounds pretty long, but I think it's okay since we would type its name in not many places.

Speaking of it, how do you think of creating a namespace (or class with static properties) to omit the long prefix?

```ts
namespace UserWhiteGaussianNoiseDataset {
  export interface Config { ... };

  export const defaultConfig: Config = { ... };

  export function create(): Dataset { ... };
}
```

Reply:
I can rename the dataset to `UserWhiteNoiseDataset` to make it shorter. What I mean is the name does not well describe the feature pattern of this dataset.

I think using `namespace` is not a good pattern in TypeScript (I remember there are some posts talking about it?), but I don't believe class with static methods is suitable for this case either. Another way is to take advantage of the module:
```typescript
// in dataset.ts
export interface config { ... };
export const defaultConfig { ... };
export function create(): Dataset { ... };

// in other ts files
import * as UserWhiteNoiseDataset from '...'
```
but it will be painful to specify the name of dataset in every ts file.

Reply:
I don't think we will directly type the dataset name in many places other than in `DataService` and `DashboardComponent`, right? I think the usage of it as in your snippet looks good enough.